lib/msal-core/src/authority/Authority.ts

```diff
@@ -147,11 +148,21 @@ export abstract class Authority {
      * Discover endpoints via openid-configuration
      * If successful, caches the endpoint for later use in OIDC
      */
-    public async resolveEndpointsAsync(telemetryManager?: TelemetryManager, correlationId?: string): Promise<Authority> {
+    public async resolveEndpointsAsync(telemetryManager?: TelemetryManager, correlationId?: string): Promise<ITenantDiscoveryResponse> {
         const openIdConfigurationEndpointResponse = await this.GetOpenIdConfigurationEndpointAsync(telemetryManager, correlationId);
         this.tenantDiscoveryResponse = await this.DiscoverEndpoints(openIdConfigurationEndpointResponse, telemetryManager, correlationId);
 
-        return this;
+        return this.tenantDiscoveryResponse;
+    }
+
+    /**
+     * Checks if there is a cached tenant discovery response with required fields.
+     */
+    public validateTenantDiscoveryResponse(): boolean {
```

Comment:
since this returns a boolean, it might be better named `hasValidTenantDiscoveryResponse`

Reply:
Or maybe `hasCachedTenantMetadata`?

Reply:
I like `hasCachedMetadata`